src/lexer/Lexer.ts

```diff
@@ -122,218 +122,206 @@ export class Lexer {
         return this.current >= this.source.length;
     }
 
+    /**
+     * Map for looking up token functions based solely upon a single character
+     * Should be used in conjunction with `tokenKindMap`
+     */
+    private static tokenFunctionMap = {
```

Comment:
This isn't the prettiest approach in the world, but as a result, it yielded much higher throughput, and it's not terribly difficult to understand, so I think it's worth the tradeoff.